client/js/widgets/widget.js

```diff
@@ -969,6 +977,80 @@ export class Widget extends StateManaged {
         };
       }
 
+      if(a.func == 'TURN') {
+        setDefaults(a, { turn: 1, turnCycle: 'inc', source: 'all', collection: 'TURN' });
+        if([ 'forward', 'backward', 'random', 'position' ].indexOf(a.turnCycle) == -1) {
+          problems.push(`Warning: turnCycle ${a.turnCycle} interpreted as forward.`);
+          a.turnCycle = 'forward'
+        }
+        //copied from select
+        let c = (a.source == 'all' ? Array.from(widgets.values()) : collections[a.source]).filter(w=>w.get('type')=='seat');
+
+        //this get the list of valid index
+        const indexList = []
+        let turn = 1
+        for(const w of c) {
+          if(indexList.indexOf(w.get('index')) == -1 && w.get('player'))
+            indexList.push(w.get('index'));
+          if(w.get('turn'))
+            turn = w.get('index')
+        }
+
+        //loop so it goes for the n next valid index
+        for(let i = 0; i < Math.abs(a.turn) % indexList.length; ++i) {
+          //this checks the next valid index
+          if((a.turnCycle == 'forward' && a.turn > 0) || (a.turnCycle == 'backward' && a.turn < 0)) {
+            indexList.sort((a,b)=>a-b);
+            if(turn >= indexList[indexList.length-1]) {
+              turn = indexList[0];
+            } else {
+              for(const idx of indexList){
+                if(idx > turn){
+                  turn = idx;
+                  break;
+                }
+              }
+            }
+          } else if((a.turnCycle == 'forward' && a.turn < 0) || (a.turnCycle == 'backward' && a.turn > 0)) {
+            //this checks the previous valid index
+            indexList.sort((a,b)=>b-a);
+            if(turn <= indexList[indexList.length-1]) {
+              turn = indexList[0];
+            } else {
+              for(const idx of indexList) {
+                if(idx < turn){
+                  turn = idx;
+                  break;
+                }
+              }
+            }
+          }
+        }
+
+        if(a.turnCycle == 'position') {
+          indexList.sort((a,b)=>a-b);
+          if(a.turn == 'first') {
+            turn = indexList[0];
+          } else if(a.turn == 'last') {
+            turn = indexList[indexList.length];
+          } else if(a.turn < 1) {
+            turn = indexList[indexList.length - ((Math.abs(a.turn - 1) % indexList.length) || indexList.length)];
+          } else {
+            turn = indexList[(a.turn - 1) % indexList.length] || 0;
+          }
+        }
+        if(a.turnCycle == 'random')
+          turn = indexList[Math.floor(Math.random() * indexList.length)];
+
+        collections[a.collection] = [];
+        //saves turn into all seats and creates output collection with turn seats
+        for(const w of c) {
+          await w.set('turn', w.get('index') == turn);
+          if(w.get('turn') == w.get('index') && w.get('player'))
```

Comment:
These two lines seem weird to me. You set `turn` to be a boolean and in the next line you compare it to `index` (which I assume is a number). Comparing `false == 0` or `true == 1` will work but I don't think it does what you want.

My guess is that you forgot to change this when you changed `turn` to be a boolean.

Reply:
you are right I did forget